src/components/DashboardHome/DashboardHome.jsx

```diff
@@ -13,124 +14,147 @@ import Sidebar from './Sidebar';
-                <FontAwesomeIcon className="mr-2" icon={faExclamationTriangle} />
-              </div>
-              <div>
-                You are not authorized to view this page.
-                This page is reserved for masters students only.
-                You may access public edX courses on edx.org.
-                If you are a masters student and believe you should have access,
-                please contact your advisor at the university for further assistance.
-              </div>
+  renderError = () => (
+    <div className="container my-4">
+      <StatusAlert
+        alertType="danger"
+        dialog={
+          <div className="d-flex">
+            <div>
+              <FontAwesomeIcon className="mr-2" icon={faExclamationTriangle} />
             </div>
-            }
-          dismissible={false}
-          open
-        />
-      </div>
-    );
-  }
+            <div>
+              You are not authorized to view this page.
+              This page is reserved for Masters students only.
+              You may access public edX courses on
+              {' '}
+              <a className="alert-link" href="https://edx.org">edX.org</a>.
+              If you are a Masters student and believe you should have access,
+              please contact your advisor at the university for further assistance.
+            </div>
+          </div>
+          }
+        dismissible={false}
+        open
+      />
+    </div>
+  );
 
   render() {
+    const { hasProgramAccess } = this.state;
+    const { pageContext, isLoading } = this.props;
+    const { programUUID, programName } = pageContext;
+
     return (
       <IntlProvider locale="en">
         <Layout>
-          {
-           this.state.validAccess ? (
-             <>
-               <Helmet title="Master's Degree in Analytics" />
-               <main id="content">
-                 <Hero
-                   programTitle="Master's Degree in Analytics"
-                   organizationLogo={{
-            url: 'https://www.edx.org/sites/default/files/school/image/logo/gtx-logo-200x101.png',
-            alt: 'Georgia Tech Institute of Technology logo',
-          }}
-                   textureImage="https://prod-discovery.edx-cdn.org/media/degree_marketing/campus_images/gt-cyber-title_bg_img_440x400.jpg"
-                   coverImage="https://prod-discovery.edx-cdn.org/media/degree_marketing/campus_images/gt_cyber_campus_image_1000x400.jpg"
-                 />
-                 <div className="container py-5">
-                   <div className="row">
-                     <div className="col-xs-12 col-lg-7">
-                       <MainContent />
-                     </div>
-                     <MediaQuery minWidth={breakpoints.large.minWidth}>
-                       {matches => matches && (
-                       <aside className="col offset-lg-1">
-                         <Sidebar />
-                       </aside>
+          {isLoading ? (
```

Comment:
The code flow is as follows:
* On mount, fetch data from the API.
* Is API currently fetching? Show loading indicator.
* API done loading data?
  * Is the user enrolled in a valid program? If so, show the dashboard content. Otherwise, show unauthorized error.
* If the API request responds with an error, there will be no valid programs and the unauthorized error will show.